lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -311,95 +298,49 @@ define((require, exports, module) ->
-        if not tokenIterator.getCurrentToken()?
-          return null
-      else
-        # following loop pushes tokenIterator to the end of
-        # boundary sequence, if it is inside one
-        # The loop isn't executed, if current token is null, because:
-        #   a) we don't need to -- if `tokenIterator` is initially on
-        #      the empty line, then it is guaranteed not to be inside boundary sequence
-        #   b) it can cause bugs -- if we are looking for start sequence and
-        #      `tokenIterator` is initially before the start of a document,
-        #      current token is null, and after stepping forward `tokenIterator` is on the
-        #      start of a document. If start sequence happens to be there, then equation start is
-        #      then found without any problem, whereas in this case null should be returned
-        for boundarySequence in boundarySequences
-          for token in boundarySequence.slice(0).reverse()
-            curToken = tokenIterator.getCurrentToken()
-            if curToken? and equalTokens(token, curToken)
-              moveFromBoundary()
-
-      tokenSequenceFinder = new TokenSequenceFinder(boundarySequences)
-      while true
-        moveToBoundary()
-        curToken = tokenIterator.getCurrentToken()
-        if not curToken?
-          return null
-
-        tokenSequenceFinder.progressIndices(curToken)
-
-        maybeFinishedSequenceId = tokenSequenceFinder.getMaybeFinishedSequenceId()
-        if maybeFinishedSequenceId?
-          { row: curTokenRow, column: curTokenColumn } = tokenIterator.getCurrentTokenPosition()
-          curTokenLength = curToken.value.length
-          finishedSequence = boundarySequences[maybeFinishedSequenceId]
-
-          equationOuterBoundary = {
-            row: curTokenRow
-            column: curTokenColumn + (if start then 0 else curTokenLength)
-          }
-
-          # after finding boundary, tokenIterator is exactly on the last token
-          # of ending sequence, so we step back until we're on the last token
-          # to determine inner boundary
-          for i in [0...finishedSequence.length-1]
-            moveFromBoundary()
-
-          { row: curTokenRow, column: curTokenColumn } = tokenIterator.getCurrentTokenPosition()
-          curTokenLength = tokenIterator.getCurrentToken().value.length
-
-          equationInnerBoundary = {
-            row: curTokenRow
-            column: curTokenColumn + (if start then curTokenLength else 0)
-          }
-
-          # to move inside the sequence, we move token iterator one more time
-          moveFromBoundary()
-
-          return {
-            id: maybeFinishedSequenceId
-            outer: equationOuterBoundary
-            inner: equationInnerBoundary
-          }
+      currentToken = tokenIterator.getCurrentToken()
+      { row: prevRow } = tokenIterator.getCurrentTokenPosition()
```

Comment:
again, it seems to be just a clumsy way of writing `prevRow = tokenItrator.getCurrentTokenPosition().row`, no?